water-gui.py

```diff
@@ -135,8 +135,9 @@ def register_page(self) -> None:
         setSize(self.root, 489, 280)
         # data variables to prompt
         self.reading = DoubleVar()
-        self.client = StringVar()
+        self.menu_selection = IntVar()
         self.per_unit = ProductMap()
+        self.product_index = list(self.per_unit)

```

Comment:
Hmm... The `__list__` method has not been set up for `ProductMap`.

I'm too lazy to open a new pull request, so I'll commit here later :smile:.

Reply:
What we must do is adding a `__list__` method for `tools.database.ProductMap`:

```python
def __list__(self) -> List[str]:
    return [...] # whatever we'll include here
```

Reply:
I made that on commit bfd0c86.